File: 120812.py
"""최빈값구하기"""
def solution(array):
    count_dict = {}
    answer = 0
    answer_count = 0 
    for arr in list(set(array)):
        count_dict[arr] = array.count(arr)
    
    for key in count_dict.keys():
        if count_dict[key] > answer_count:
            answer = key
            answer_count = count_dict[key]
        elif count_dict[key] == answer_count:
            answer = -1
    return answer

print(solution([1, 2,2, 2, 3, 3, 3,4,4,4,4]))

# 다른사람풀이
def solution(array):
    while len(array) != 0:
        for i, a in enumerate(set(array)):
            array.remove(a)
        if i == 0: return a
    return -1

File: 120812.js
// 최빈값 구하기
function solution(array) {
  let cntNumber = {};
  for (let arr of array) {
    if (!(arr in cntNumber)) {
      cntNumber[arr] = 1;
    } else {
      cntNumber[arr] += 1;
    }
  }
  let maxNumber = 0;
  let maxCnt = 0;
  for (let i of Object.keys(cntNumber)) {
    if (cntNumber[i] > maxCnt) {
      maxNumber = i;
      maxCnt = cntNumber[i];
    } else if (cntNumber[i] === maxCnt) {
      maxNumber = -1;
    }
  }
  return maxNumber;
}
console.log(solution([1, 2, 3, 3, 3, 4]));
console.log(solution([1, 1, 2, 2]));
